devsetup/scripts/bmaas/sushy-emulator.sh

```diff
@@ -18,26 +18,36 @@ MY_TMP_DIR="$(mktemp -d)"
 trap 'rm -rf -- "$MY_TMP_DIR"' EXIT
 
 NODE_NAME_PREFIX=${NODE_NAME_PREFIX:-"crc-bmaas"}
-LIBVIRT_USER=${LIBVIRT_USER:-"sushyemu"}
 NAMESPACE=${SUSHY_EMULATOR_NAMESPACE:-"sushy-emulator"}
-SSH_ALGORITHM=rsa
-SSH_KEY_FILE=bmaas-ssh-key-id_rsa
-SSH_KEY_SIZE=4096
-CRC_NETWORK_NAME=crc
+DRIVER=${SUSHY_EMULATOR_DRIVER:-"libvirt"}
 INGRESS_DOMAIN=$(oc get ingresses.config/cluster -o jsonpath={.spec.domain})
 REDFISH_USERNAME=${REDFISH_USERNAME:-"admin"}
 REDFISH_PASSWORD=${REDFISH_PASSWORD:-"password"}
-if sudo systemctl is-active libvirtd.service; then
-    LIBVIRT_SOCKET="?socket=/var/run/libvirt/libvirt-sock"
-else
-    LIBVIRT_SOCKET=""
-fi
 IMAGE=${SUSHY_EMULATOR_IMAGE:-"quay.io/metal3-io/sushy-tools:latest"}
 
-# TODO: Make CRC_NETWORK_NAME a parameter so that this script can be used on
-# separate hypervisor, against any OCP.
-LIBVIRT_IP_ADDRESS=$(nmcli connection show "${CRC_NETWORK_NAME}" | grep ipv4.addresses | cut -d / -f 1 | awk '{ print $2 }')
-INSTANCES=$(virsh --connect=qemu:///system list --all --uuid --name | grep "${NODE_NAME_PREFIX}" | awk '{ printf "\"" $1 "\", " }')
+if [ $DRIVER = "libvirt" ]; then
+    CRC_NETWORK_NAME=crc
+    SSH_ALGORITHM=rsa
+    SSH_KEY_FILE=bmaas-ssh-key-id_rsa
+    SSH_KEY_SIZE=4096
+    LIBVIRT_USER=${LIBVIRT_USER:-"sushyemu"}
+    if sudo systemctl is-active libvirtd.service; then
+        LIBVIRT_SOCKET="?socket=/var/run/libvirt/libvirt-sock"
+    else
+        LIBVIRT_SOCKET=""
+    fi
+    # TODO: Make CRC_NETWORK_NAME a parameter so that this script can be used on
+    # separate hypervisor, against any OCP.
+    LIBVIRT_IP_ADDRESS=$(nmcli connection show "${CRC_NETWORK_NAME}" | grep ipv4.addresses | cut -d / -f 1 | awk '{ print $2 }')
+    INSTANCES=$(virsh --connect=qemu:///system list --all --uuid --name | grep "${NODE_NAME_PREFIX}" | awk '{ printf "\"" $1 "\" " }' | sed -e 's/" "/", "/' -e 's/^"/["/' -e 's/" $/"]/')
+    LIBVIRT_URI="'qemu+ssh://${LIBVIRT_USER}@${LIBVIRT_IP_ADDRESS}/system${LIBVIRT_SOCKET}'"
+    EMULATOR_OS_CLOUD="None"
+elif [ $DRIVER = "openstack" ]; then
+    OS_CLIENT_CONFIG_FILE=${SUSHY_EMULATOR_OS_CLIENT_CONFIG_FILE:-/etc/openstack/clouds.yaml}
+    INSTANCES=$(openstack --os-cloud=${SUSHY_EMULATOR_OS_CLOUD} server list --name "edpm-compute.*" -f json -c ID | jq -c [.[].ID])
+    LIBVIRT_URI="None"
+    EMULATOR_OS_CLOUD="'${SUSHY_EMULATOR_OS_CLOUD}'"
+fi
```

Comment:
We may want to add a final else here, or earlier in the script and error if $DRIVER is not one of the supported drivers.